libs/extensions/std/exec/src/gtfs-rt-interpreter-executor.ts

```diff
@@ -0,0 +1,159 @@
+    if (E.isLeft(sheet)) {
+      return Promise.resolve(
+        R.err({
+          message: sheet.left.message,
+          diagnostic: { node: context.getCurrentNode(), property: 'name' },
+        }),
+      );
+    }
+    return R.ok(sheet.right);
+  }
+
+  private async parseFeedMessage(
+    entity_type: string,
+    feedMessage: GtfsRealtimeBindings.transit_realtime.FeedMessage,
+    context: ExecutionContext,
+  ): Promise<Either<Error, Sheet>> {
+    return new Promise((resolve) => {
+      switch (entity_type) {
+        case 'trip_update':
+          // Extract the trip updates from thee feed message
+          resolve(this.parseTripUpdates(feedMessage, context));
+          break;
+        case 'alert':
+          // TODO: parse alert;
+          break;
+        case 'vehicle':
+          // TODO: parse vehicle;
+          break;
+
+        // No entitiy detected -> error
+        default:
+          resolve(E.left(new Error('No entity detected in GTFS-RT-Feed. ')));
+          break;
+      }
+    });
+  }
+
+  private parseTripUpdates(
+    feedMessage: GtfsRealtimeBindings.transit_realtime.FeedMessage,
+    context: ExecutionContext,
+  ): Promise<Either<Error, Sheet>> {
+    return new Promise((resolve) => {
+      context.logger.logDebug(`Parsing raw data as TripUpdates"`);
+      const header: string[] = Object.keys({} as TripUpdates);
+      const rows: string[][] = [];
+      rows.push(header);
+      for (const entity of feedMessage.entity) {
+        if (entity.tripUpdate) {
+          if (entity.tripUpdate.stopTimeUpdate) {
+            for (const stop_time_update of entity.tripUpdate.stopTimeUpdate) {
+              const tripUpdate: TripUpdates = {
+                'header.gtfs_realtime_version':
+                  feedMessage.header.gtfsRealtimeVersion,
+                'header.timestamp': String(feedMessage.header.timestamp),
+                'header.incrementality': String(
+                  feedMessage.header.incrementality,
+                ),
+                'entity.id': String(entity.id),
+                'entity.trip_update.trip.trip_id': String(
+                  entity.tripUpdate.trip.tripId,
+                ),
+                'entity.trip_update.trip.route_id': String(
+                  entity.tripUpdate.trip.routeId,
+                ),
+                'entity.trip_update.stop_time_update.stop_sequence': String(
+                  stop_time_update.stopSequence,
+                ),
+                'entity.trip_update.stop_time_update.stop_id': String(
+                  stop_time_update.stopId,
+                ),
+                'entity.trip_update.stop_time_update.arrival.time': String(
+                  stop_time_update.arrival?.time,
+                ),
+                'entity.trip_update.stop_time_update.departure.time': String(
+                  stop_time_update.departure?.time,
+                ),
+              };
+              rows.push(Object.values(tripUpdate) as string[]);
```

Comment:
There is no need to build the object with string keys if you just remove them here again.

Reply:
For easier maintainment of code i decided to continue using the object with keys. Easier to handle and no need to take care of having the right index.